server/lib/auth0.js

```diff
@@ -88,7 +127,7 @@ export const deleteRules = (progress, client, rules) => {
     return Promise.resolve(true);
   }
 
-  progress.log('Deleting rules that no longer exist in the repository...');
+  progress.log('Deleting all rules to replace them with the ones existing in the repository...');
```

Comment:
You are right. this is for a couple of reasons. 
1. Simplify rule deployment avoiding having to deal with order, stages and some other unexpected consequences.
2. Avoid inconsistent states. What would happen if one of the new rules expects a value from other rule that has not been applied yet? My guess its that rules will fail going back to the scenario you mentioned initially.

Other option I looked into is reverse the order. Create new rules and when all are created successfully remove the old ones but this may also have unexpected consequences during deployment